### PR TITLE
[CN-962] Upgrade operator-sdk version to 1.33.0

### PR DESCRIPTION
**Pull request by semihbkgr**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,14 +19,14 @@ TOOLBIN = $(shell pwd)/bin
...
 # https://github.com/operator-framework/operator-sdk/releases
-OPERATOR_SDK_VERSION ?= v1.28.1
+OPERATOR_SDK_VERSION ?= v1.33.0
 # https://github.com/kubernetes-sigs/controller-tools/releases
-CONTROLLER_GEN_VERSION ?= v0.11.3
+CONTROLLER_GEN_VERSION ?= v0.12.0
 # https://github.com/kubernetes-sigs/controller-runtime/releases
 # It is set in the go.mod file
-CONTROLLER_RUNTIME_VERSION ?= v0.13.1
+CONTROLLER_RUNTIME_VERSION ?= v0.15.0
```
Comment (kutluhanmetin) [inline Makefile:29]: Shouldn't it be `v0.15.2`

File: .github/workflows/docs-release.yaml [hunk 2]
```diff
@@ -70,7 +70,7 @@ jobs:
       - name: Set up Golang
         uses: actions/setup-go@v5
         with:
-          go-version: "1.21.0"
+          go-version: "1.20.0"
```
Comment (kutluhanmetin) [inline .github/workflows/docs-release.yaml:73]: I just wonder why did we have to lower go version?

File: Makefile (hunk 1 above)
Comment (semihbkgr) [reply Makefile:29]: I realized that this variable is not even in use.

File: .github/workflows/docs-release.yaml (hunk 2 above)
Comment (semihbkgr) [reply .github/workflows/docs-release.yaml:73]: The `go` version from `operator-sdk` v1.33 is `1.20`

**Pull request merged**